edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Amqp/CbsNode.cs

```diff
@@ -185,48 +199,51 @@ internal static (string deviceId, string moduleId) ParseIds(string audience)
         // Note: This method updates this.clientCredentialsMap, and should be invoked only within this.identitySyncLock
         internal async Task<(AmqpResponseStatusCode, string)> UpdateCbsToken(AmqpMessage message)
         {
-            IClientCredentials clientCredentials;
+            ClientToken clientToken;
+
             try
             {
-                clientCredentials = this.GetClientCredentials(message);
+                clientToken = this.GetClientToken(message);
             }
             catch (Exception e) when (!ExceptionEx.IsFatal(e))
             {
                 Events.ErrorGettingIdentity(e);
                 return (AmqpResponseStatusCode.BadRequest, e.Message);
             }
 
-            if (!this.clientCredentialsMap.TryGetValue(clientCredentials.Identity.Id, out CredentialsInfo credentialsInfo))
-            {
-                this.clientCredentialsMap[clientCredentials.Identity.Id] = new CredentialsInfo(clientCredentials);
-            }
-            else
+            // Insert/update the cached tokens with the new value
+            this.clientTokens[clientToken.Id] = clientToken;
+
+            if (this.clientTokens.TryGetValue(clientToken.Id, out ClientToken credentialsInfo))
```

Comment:
? You are adding the token and then immediately checking if the token exists?
Here we should just always add the token.

Reply:
Forgot to remove the extra check when I split out the authentication state and the token.  The intent is to call into the credcache if we've already authenticated this token.

---
In reply to: [489204250](https://github.com/Azure/iotedge/pull/3552#discussion_r489204250) [](ancestors = 489204250)

Reply:
Talked offline, filed another PBI to come back and fix the cred-cache usage.